marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.tsx

```diff
@@ -500,27 +498,18 @@ const MappingCard: React.FC<Props> = (props) => {
                 }
                 nestedEntityData.push(propty);
                 extractNestedEntityData(val.subProperties, propty.children, parentKey);
-
-                if(parentKey !== tempKey){
-                    parentKey = tempKey;
-                }
+                parentKey = (parentKey.indexOf("/")!=-1)?parentKey.substring(0,parentKey.lastIndexOf('/')):''
```

Comment:
Is this based on the assumption that the entity title is the part of an entityTypeId that comes after the last slash? Would it be possible for the backend to provide this information instead? 

I'm leery of the UI having any knowledge of how to parse bits out of the entityTypeId. I'd like to consolidate all of that logic into entity-lib.sjs in the backend so that if any assumptions prove incorrect, we only have to change it in a single place in all of DHF.

Reply:
No, this doesn't have anything to do with entityTypeId. This method converts the nested entity properties to array that is needed for entity table in the ui . The keys of the table are unique and it is built by concatenating the property names (name/firstName/prefix etc)..